operator/pkg/apis/infrastructure/v1alpha1/controlplane.go

```diff
@@ -41,47 +42,47 @@ type ControlPlaneList struct {
 // master and etcd are configured to run. By default, KIT uses all the default
 // values and ControlPlaneSpec can be empty.
 type ControlPlaneSpec struct {
-	Infrastructure    `json:",inline"`
 	KubernetesVersion string     `json:"kubernetesVersion,omitempty"`
 	Master            MasterSpec `json:"master,omitempty"`
 	Etcd              ETCDSpec   `json:"etcd,omitempty"`
 }
 
-type Infrastructure struct {
-	VPCCidr        string   `json:"vpccidr,omitempty"`
-	Zones          []string `json:"zones,omitempty"`
-	PrivateSubnets []string `json:"privateSubnets,omitempty"`
-}
-
 // MasterSpec provides a way for the user to configure master instances and
 // custom flags for components running on master nodes like apiserver, KCM and
 // scheduler.
 type MasterSpec struct {
 	Instances         `json:",inline"`
-	Scheduler         *Config `json:"scheduler,omitempty"`
-	ControllerManager *Config `json:"controllerManager,omitempty"`
-	APIServer         *Config `json:"apiServer,omitempty"`
+	Scheduler         *Component `json:"scheduler,omitempty"`
+	ControllerManager *Component `json:"controllerManager,omitempty"`
+	APIServer         *Component `json:"apiServer,omitempty"`
 }
 
 // ETCDSpec provides a way to configure the etcd nodes and args which are passed to the etcd process.
 type ETCDSpec struct {
-	Instances `json:",inline"`
-	*Config   `json:",inline"`
+	Instances  `json:",inline"`
+	*Component `json:",inline"`
 }
 
-// Config provides a generic way to pass in args and images to master and etcd
+// Component provides a generic way to pass in args and images to master and etcd
 // components. If a user wants to change the QPS they need to provide the
 // following flag with the desired value -`kube-api-qps:100` in the args.
-type Config struct {
-	Args  map[string]string `json:"args,omitempty"`
-	Image string            `json:"image,omitempty"`
+type Component struct {
+	Replicas int         `json:"replicas,omitempty"`
+	Spec     *v1.PodSpec `json:"spec,omitempty"`
 }
 
 // Instances denotes how the infrastructure of a particular components looks
 // like, if a user wants to use a specific AMI ID, they can provide this in the
 // Instances for the corresponding component.
 type Instances struct {
-	AMI   string `json:"ami,omitempty"`
-	Type  string `json:"type,omitempty"`
-	Count int    `json:"count,omitempty"`
+	AMI  string `json:"ami,omitempty"`
+	Type string `json:"type,omitempty"`
+}
+
+func (c *ControlPlane) ClusterName() string {
+	return c.Name
```

Comment:
do we have Name as part of ControlPlane spec ? I don't remember seeing it unless I'm misreading this.

Reply:
Yes, its part of the `metadata` field